streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -155,7 +156,8 @@ public void process(final Record<K, V1> record) {
                         // we may delete some values with the same key early but since we are going
                         // range over all values of the same key even after failure, since the other window-store
                         // is only cleaned up by stream time, so this is okay for at-least-once.
-                        store.putIfAbsent(TimestampedKeyAndJoinSide.make(!isLeftSide, record.key(), otherRecordTimestamp), null);
+                        final JoinSide otherJoinSide = joinSide.isLeftSide() ? JoinSide.RIGHT : JoinSide.LEFT;
```

Comment:
This could be a `JoinSide#complement` or `JoinSide#opposite` function, in case there are other situations where we need to know what the other side of the join is doing.